app/src/main/java/org/wikipedia/gallery/GalleryActivity.java

```diff
@@ -546,6 +546,11 @@ private void loadGalleryContent() {
     }
 
     private void applyGalleryList(@NonNull List<MediaListItem> list) {
+        if (list.isEmpty()) {
```

Comment:
This doesn't really fix the root of the problem, which is that we're failing to load an image if it's not listed in the article's media-list endpoint.

See my other fix in #587, which should fix this issue indirectly.

Reply:
Yes, your PR works correctly!